process/unDelegateDelegatorsLegacy.go

```diff
@@ -0,0 +1,122 @@
+package process
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/multiversx/mx-chain-tools-accounts-manager-go/core"
+	"github.com/multiversx/mx-chain-tools-accounts-manager-go/data"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+const getUserStakeByTypeEndpoint = "getUserStakeByType"
+
+func (ag *accountsGetter) putUnDelegatedValuesFromDelegationLegacy(accountsWithStake map[string]*data.AccountInfoWithStakeValues) error {
+	unDelegatedValues, err := ag.getUnDelegatedValuesFromDelegationLegacyContract(accountsWithStake)
+	if err != nil {
+		return err
+	}
+
+	for address, unDelegatedValue := range unDelegatedValues {
+		if _, found := accountsWithStake[address]; !found {
+			continue
+		}
+
+		accountsWithStake[address].UnDelegateLegacy = unDelegatedValue.String()
+		accountsWithStake[address].UnDelegateLegacyNum = core.ComputeBalanceAsFloat(unDelegatedValue.String())
+	}
+
+	return nil
+}
+
+func (ag *accountsGetter) getUnDelegatedValuesFromDelegationLegacyContract(accountsWithStake map[string]*data.AccountInfoWithStakeValues) (map[string]*big.Int, error) {
+	defer logExecutionTime(time.Now(), "Fetched undelegated values from delegation legacy contract")
+
+	unDelegatedValue := make(map[string]*big.Int)
+
+	done, wg := make(chan struct{}, maxNumberOfParallelRequests), &sync.WaitGroup{}
+	errors := make([]string, 0)
+	for address := range accountsWithStake {
+		done <- struct{}{}
+		wg.Add(1)
+
+		go func(addr string) {
+			value, err := ag.getUnDelegatedValueForAddressDelegationLegacyContract(addr, done, wg)
+			if err != nil {
+				ag.mutex.Lock()
```

Comment:
the mutex of `accountsGetter` does not make too much sense here, since you only protect local variables. suggestion to declare a new mutex inside this function, so multiple functions can be run in parallel.